codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestXmlProtocolGenerator.java

```diff
@@ -20,6 +24,15 @@
         super(true);
     }
 
+    @Override
+    public void generateProtocolTests(GenerationContext context) {
+        new HttpProtocolTestGenerator(context,
+                new HttpProtocolUnitTestRequestGenerator.Builder(),
+                new HttpProtocolUnitTestResponseGenerator.Builder(),
+                new HttpProtocolUnitTestResponseErrorGenerator.Builder()
+        ).generateProtocolTests();
+    }
```

Comment:
probably want to also add the options to the buildsers. It probably makes sense for the ConfigOptions to be a shared utility that can be used by multiple protocols, since I think it will be pretty generic across them all.

Reply:
Rebasing with this util should help, https://github.com/aws/aws-sdk-go-v2/pull/599/files#diff-81289b07c59dbc0f9adcdd205eac2266R38-R46

Reply:
Updated to use AwsProtocolUtils